File: src/main/java/net/minestom/server/permission/PermissionVerifier.java
package net.minestom.server.permission;

import org.jetbrains.annotations.Nullable;
import org.jglrxavpok.hephaistos.nbt.NBTCompound;

/**
 * Interface used to check if the {@link NBTCompound nbt data} of a {@link Permission} is correct.
 */
@FunctionalInterface
public interface PermissionVerifier {

    /**
     * Called when using {@link PermissionHandler#hasPermission(String, PermissionVerifier)}.
     *
     * @param nbtCompound the data of the permission, can be null if not any
     * @return true if {@link PermissionHandler#hasPermission(String, PermissionVerifier)}
     * should return true, false otherwise
     */
    boolean isValid(@Nullable NBTCompound nbtCompound);
}

File: src/main/java/net/minestom/server/permission/PermissionVerifier.kt
package net.minestom.server.permission

import org.jglrxavpok.hephaistos.nbt.NBTCompound
import net.minestom.server.permission.PermissionVerifier

/**
 * Interface used to check if the [nbt data][NBTCompound] of a [Permission] is correct.
 */
fun interface PermissionVerifier {
    /**
     * Called when using [PermissionHandler.hasPermission].
     *
     * @param nbtCompound the data of the permission, can be null if not any
     * @return true if [PermissionHandler.hasPermission]
     * should return true, false otherwise
     */
    fun isValid(nbtCompound: NBTCompound?): Boolean
}